apps/opik-backend/src/main/java/com/comet/opik/domain/DatasetItemDAO.java

```diff
@@ -379,47 +388,49 @@ public Mono<Long> save(@NonNull UUID datasetId, @NonNull List<DatasetItem> items
             return Mono.empty();
         }
 
-        return inset(datasetId, items)
-                .retryWhen(AsyncUtils.handleConnectionError());
+        return inset(datasetId, items);
     }
 
     private Mono<Long> inset(UUID datasetId, List<DatasetItem> items) {
-        return asyncTemplate.nonTransaction(connection -> {
+        List<List<DatasetItem>> batches = Lists.partition(items, bulkConfig.getSize());
```

Comment:
This partitioning increases the chances of partial insertions, specially under error scenarios, and its derived problems.

Any idea on how far it is from the ClickHouse query size limit with batches of 1000s items? (for dataset item, for experiment item etc.).

Reply:
Depending on the JSON size, we can increase the query size limit, but I believe we might need this in some way. I also thought about partial inserts, but then I checked, and transactions still need to be supported in the R2DBC driver. Once they are, we could involve all transaction requests or create a proper pipeline to keep track of batch inserts.